src/pypgstac/src/lib.rs

```diff
@@ -1,83 +1,68 @@
-use serde_json::{Map, Value};
-use thiserror::Error;
-use pyo3::{create_exception, exceptions::PyException, prelude::*};
+use ::pyo3::{
+    prelude::*,
+    types::{PyDict, PyList, PyString},
+};
+use anyhow::{anyhow, Error};
 
 const MAGIC_MARKER: &str = "𒍟※";
 
-create_exception!(pgstacrs, HydrationError, PyException);
-
-pub trait Hydrate {
-    fn hydrate(&mut self, base: Self) -> Result<(), Error>;
-}
-
-#[derive(Debug, Error)]
-pub enum Error {
-    #[error("type mismatch")]
-    TypeMismatch(Value, Value),
-}
-
-impl Hydrate for Value {
-    fn hydrate(&mut self, base: Self) -> Result<(), Error> {
-        match self {
-            Value::Object(item) => match base {
-                Value::Object(base) => item.hydrate(base),
-                _ => Err(Error::TypeMismatch(self.clone(), base)),
-            },
-            Value::Array(item) => match base {
-                Value::Array(base) => item.hydrate(base),
-                _ => Err(Error::TypeMismatch(self.clone(), base)),
-            },
-            _ => Ok(()),
+#[pyfunction]
+fn hydrate<'a>(base: &PyAny, item: &'a PyAny) -> PyResult<&'a PyAny> {
+    fn hydrate_any<'a>(base: &PyAny, item: &'a PyAny) -> PyResult<&'a PyAny> {
```

Comment:
I wouldn't have usually written inner functions in the internal scope of the python function... I wonder if that's in effect just stylistic or whether there's any compilation performance that changes 🤷‍♂️